flamedisx/x1t_sr1.py

```diff
@@ -15,21 +19,126 @@
+            data_bag.append(tmp)
+    support_def = tmp['coordinate_system'][0][1] 
+    
+    return yy_ref_bag, support_def
+
+def cal_bias_tf(sig, fmap, support_def, pivot_pt):
+    tmp = tf.convert_to_tensor(sig, dtype=fd.float_type())
+    bias_low = tfp.math.interp_regular_1d_grid(x=tmp, 
+            x_ref_min=support_def[0], x_ref_max=support_def[1], y_ref=fmap[0],
+            fill_value='constant_extension')
+    bias_high = tfp.math.interp_regular_1d_grid(x=tmp, 
+            x_ref_min=support_def[0], x_ref_max=support_def[1], y_ref=fmap[1],
+            fill_value='constant_extension')
+
+    tmp = tf.math.subtract(bias_high, bias_low)
+    tmp1 = tf.math.scalar_mul(pivot_pt, tmp)
+    bias_out = tf.math.add(tmp1, bias_low)
+    bias_out = tf.math.add(bias_out, tf.ones_like(bias_out))
+
+    return bias_out
+
 ##
+# Loading combined cuts acceptances
+##
+def itp_cut_accept_tf(sig, fmap, support_def):
+    accept_out = tf.squeeze(tfp.math.interp_regular_1d_grid(x=sig, 
+            x_ref_min=support_def[0], x_ref_max=support_def[1], y_ref=fmap,
+            fill_value='constant_extension'))
+    return accept_out
+
+# Defining paths
+path_cut_accept_s1 = ['/home/peaelle42/software/bbf/bbf/data/S1AcceptanceSR1_v7_Median.json']
+path_cut_accept_s2 = ['/home/peaelle42/software/bbf/bbf/data/S2AcceptanceSR1_v7_Median.json']
 
+path_dummy_ones_s1 = [dummy_base+'dummy_ones_s1.json']
+path_dummy_ones_s2 = [dummy_base+'dummy_ones_s2.json']
 
+##
+# Flamedisx sources
+##
 class SR1Source:
-    drift_velocity = 1.335 * 1e-4   # cm/ns
+    drift_velocity = def_drift_vel
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        try:
+            # Loading combined cut acceptances
+            self.cut_accept_map_s1, self.cut_accept_support_s1 = read_bias_tf(path_cut_accept_s1)
+            self.cut_accept_map_s2, self.cut_accept_support_s2 = read_bias_tf(path_cut_accept_s2)
+            print('BBF combined cut acceptances maps loaded')
+            # Loading reconstruction bias map
+            self.recon_map_s1_tf, self.support_def_s1 = read_bias_tf(path_recon_bias_mean_s1)
+            self.recon_map_s2_tf, self.support_def_s2 = read_bias_tf(path_recon_bias_mean_s2)
+            print('BBF reconstruction bias mean maps loaded')
+        except:
+            # Loading combined cut acceptances
+            self.cut_accept_map_s1, self.cut_accept_support_s1 = read_bias_tf(path_dummy_ones_s1)
+            self.cut_accept_map_s2, self.cut_accept_support_s2 = read_bias_tf(path_dummy_ones_s2)
+            print('Dummy combined cut acceptances maps loaded')
+            # Loading reconstruction bias map
+            self.recon_map_s1_tf, self.support_def_s1 = read_bias_tf(path_dummy_zeros_s1)
+            self.recon_map_s2_tf, self.support_def_s2 = read_bias_tf(path_dummy_zeros_s2)
+            print('Dummy reconstruction bias mean maps loaded')
+
+    def recon_bias_s1(self, sig, bias_pivot_pt=def_recon_pivot):
```

Comment:
Maybe bias_offset or bias_mean_shift instead of bias_pivot_pt?

Reply:
There is a lower bound, `bias_low`, and upper bound, `bias_high`, to the reconstruction bias mean as there is an uncertainty to this bias mean due to its dependence on the waveform. The final reconstruction bias mean is calculated using a single parameter `bias_pivot_pt` to linearly tune where between this lower and upper bound, the reconstruction bias mean is.

`bias_out` = `bias_pivot_pt`*(`bias_high`-`bias_low`) + `bias_low`

So in that sense, it is like the 'pivoting point'.